### DISCUSSION TITLE
How to define behaviour for conflicts during inserts?

### DISCUSSION THREAD

**Discussion by YugoCode** [pocketbase/pocketbase#2305] [Q&A]
We can now define constrains for each collection in the UI, but how do we define the behaviour for conflicts during inserts? For example I would like to update a specific field during a conflict.

In Postgres you can do this:

```sql
INSERT INTO requests (ip, hostname, last_requested) VALUES (0, "admin", "2020-05-06 22:44:12")
ON CONFLICT (ip, hostname) DO UPDATE
SET last_requested = EXCLUDED.last_requested;
```

In SQLite you can do the same with [upsert](https://www.sqlite.org/lang_UPSERT.html).

I'm using PocketBase as a framework and add records like this:

```golang
func AddRecord(collectionName string, data map[string]any) (string, error) {

	collection, err := App.Dao().FindCollectionByNameOrId(collectionName)
	if err != nil {
		return "", err
	}

	record := models.NewRecord(collection)
	form := forms.NewRecordUpsert(App, record)
	form.LoadData(data)

	if err := form.Submit(); err != nil {
		return "", err
	}
	return record.Id, nil
}
```

I don't see a parameter I could pass to `NewRecordUpsert`, which allows me to specify how to handle conflicts. **What is the preferred way to achieve what I have described?**

**And could it maybe be possible in the future do define the conflict behaviour in the UI next to defining the constrains?**

Thank you very much! :)

**Comment by dschissler** [comment]
Maybe use an update and insert event to resolve it with code logic.

https://pocketbase.io/docs/event-hooks/#db-hooks
